bodhi/client/__init__.py

```diff
@@ -43,6 +43,48 @@ def _warn_if_url_and_staging_set(ctx, param, value):
     return value
 
 
+def _validate_query_user(ctx, param, value):
+    """
+    Callback used on the --user flag on query commands (query overrides and query updates)
+    to check if both --user and --mine are set, and return a warning message to stderr
+
+    Args:
+        ctx (click.core.Context): The Click context, used to find out if the --mine flag is set.
+        param (click.core.Option): The option being handled. Unused.
+        value (unicode): The value of the --user flag.
+
+    Returns:
+        unicode: The value of the --user flag.
+    """
+    if value and ctx.params.get('mine'):
+        click.echo('Warning: --mine and --user options are both set. --user will be ignored.\n',
+                   err=True)
+    return value
+
+
+def _validate_query_mine(ctx, param, value):
+    """
+    Callback used on the --mine flag on query commands (query overrides and query updates)
+    to check if the USERNAME environment variable is set. If it isn't set, raise a
+    click.BadParameter exception.
+
+    Args:
+        ctx (click.core.Context): The Click context, used to find out if the --user flag is set.
+        param (click.core.Option): The option being handled. Unused.
+        value (Boolean): True if the --mine flag was given.
+
+    Returns:
+        Boolean: The value of the --mine flag.
+    """
+    if value and not os.environ['USERNAME']:
+        raise click.BadParameter('USERNAME environment variable not set.')
```

Comment:
I recently created an [```_init_username()```](https://github.com/fedora-infra/bodhi/blob/2.7.0/bodhi/client/bindings.py#L335-L363) method that can be used to get the user's username automatically. It uses their saved cookies to find if they've ever authenticated before and uses that name if they have. Otherwise, it falls back to prompting the user. The environment variable will override what this does since it will set username when the BodhiClient is initialized.

What if we make that a public method (by renaming it from ```_init_username``` to ```init_username```) and use that to ensure that the username will work as people expect? This way, they don't have to set an environment variable (though they still can).